src/command.ts

```diff
@@ -109,6 +112,21 @@ export class CommandBuilder implements PromiseLike<CommandResult> {
     return parseAndSpawnCommand(this.#getClonedState());
   }
 
+  /**
+   * Adds support for handling custom commands.
+   *
+   * Custom commands must be uniquely named, and cannot override any built-ins.
+   */
+  handle(command: string, handleFn: CustomCommandHandler) {
```

Comment:
I think `registerCommand` then `CustomCommandHandler` -> `CommandHandler` might be better? In the future, maybe refactoring regular commands to use the `CommandHandler` might be beneficial.

Reply:
> I think `registerCommand` then `CustomCommandHandler` -> `CommandHandler` might be better? In the future, maybe refactoring regular commands to use the `CommandHandler` might be beneficial.

Yeah I was thinking about that as well; or even registering the built-ins the same way so there's no actual difference handling-wise.  Should we do that now rather than later?

Reply:
Sure, if you'd like to! If we do that, maybe we could also add an `unregisterCommand(name: string)` method too.

Reply:
Now that I am looking at this again this morning, do we want to restrict the name of the command at all?  e.g. no path separators?

Reply:
Good point. Yeah, I think we should restrict the name.